lib/datasource/bitbucket-tags/index.ts

```diff
@@ -1,133 +1,133 @@
 import * as utils from '../../platform/bitbucket/utils';
-import * as packageCache from '../../util/cache/package';
+import { cache } from '../../util/cache/package/decorator';
 import { BitbucketHttp } from '../../util/http/bitbucket';
 import { ensureTrailingSlash } from '../../util/url';
+import { Datasource } from '../datasource';
 import type { DigestConfig, GetReleasesConfig, ReleaseResult } from '../types';
 import { BitbucketCommit, BitbucketTag } from './types';
 
-const bitbucketHttp = new BitbucketHttp();
+export class BitBucketTagsDatasource extends Datasource {
+  bitbucketHttp = new BitbucketHttp();
```

Comment:
@JamieMagee I think we should make `Datasource` more generic to allow override `http` property. So we don't have two instances of `Http` class. 🤔

Reply:
New issue?

Reply:
Something [like this](https://www.typescriptlang.org/play?#code/IYIwzgLgTsDGEAJYBthjAgIsCaD2ArlLAKYA8AKgiQB4QkB2AJhgBIQQAOCAvAu1wB8AWABQAbzEJpCTlDz14JJkjwNIUAvDxQAFJwIhkAS1gIoJYEzXIAngmNMAXAg3GGAcwA0CCLc4kLroMJADuCLqOLm6eAJS8gggUsfGSojIZvgAWxmAAdFkc3Hwh4X4BkUyxANxSMgC+YnXShVwuFLWijaJiKGhsRQhpGQZGpqrq0FoQOvqGJmYWVjb2Ua7Q7h6x4t3dfegIAELGECBaANYkEALctPTMA1xD3b2oB8enF1fYuGCExCRqHRGCwsDh8ERSGQPmdYJdrkVEsMZM1ZPNxpAcOMltYGHYHCo+AByEAnWHwomdDKo2BqDTTWapVEZMAEAJ6GFfCA-CEAvKOHycuFXG6xVG7JqifYYACCDAUWRIUB5f0hgLuIIwKv+pCGqNGC3WWMWllx+McvAQROA8ogiqglNRNLpU20eiZ6Uy0lZ7N0coVSu1av5TB8ovFYnqQA)?

I'm not sure how to set a default value for the type parameter this way.

Reply:
Yeah. Can we make `Http` class the default value if not passed?

Reply:
Created #10991 to summarise this discussion, and marking this thread as resolved for now.